### PR TITLE
feat(GitHub): add privacy mode

### LINKED ISSUE #6346: Merged PRs show wrong author name on GitHub
### Steps to Reproduce

1. Open a merged PR on GitHub
2. Observe what the presence displays on Discord

### Expected Behavior

The name of the author of the PR is displayed

### Actual Behavior

The name of person who merged the PR is displayed

### Page URL

https://github.com/discordjs/discord.js/pull/8089 (example)

### Screenshots

_No response_

### Operating System

Windows 11

### Browser and Version

Google Chrome Version 102.0.5005.115 (Official Build) (64-bit)

### Additional Comments

This happens because, under the title of the PR, the name of the author is displayed while the PR is open, however, this changes to the name of the person who merged once it gets merged. The name of the author should be taken from the `author` class instead in the first comment on the PR

### LINKED ISSUE #6355: GitHub presence fails to parse new issues page
### Steps to Reproduce

1. Go to a repo's issues tab and click "New issue"
2. Whether the repo has an issue template, issue form, or none at all, you will see the error attached below

### Expected Behavior

The presences displays when you're creating a new issue properly

### Actual Behavior

The presence always assumes that a page including /issues/ is an issue

### Page URL

https://github.com/PreMiD/Presences/issues/new/choose (for example)

### Screenshots

image

### Operating System

Windows 11

### Browser and Version

Google Chrome Version 102.0.5005.115 (Official Build) (64-bit)

### Additional Comments

When fixing this, the various types of issue forms should be accounted for. The faulty lines are the following: https://github.com/PreMiD/Presences/blob/5e5f99668ba4800524e66c2a3fc9fd38755025c6/websites/G/GitHub/presence.ts#L111-L120

### LINKED ISSUE #6357: GitHub should support features page
### Presence name

GitHub

### Description

The https://github.com/features page is currently not supported in the presence and simply displays "GitHub" with no other text

### Alternatives

_No response_

### Additional context

_No response_

### LINKED ISSUE #6435: GitHub Gist
### Presence name

GitHub: https://github.com/PreMiD/Presences/tree/main/websites/G/GitHub

### Description

Gist is missing: https://gist.github.com/

### Alternatives

N/A

### Additional context

N/A

### PR DESCRIPTION
**Pull request by realtofuine**
## Description 
add privacy mode to the GitHub Presence

closes #6435 

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

image

image

image

for #6355 
image

for #6357
image

image

for #6346
image

for #6435 
image

image

image

image

</details>

### CODE REVIEW

File: websites/G/GitHub/dist/metadata.json [hunk 1]
```diff
@@ -26,16 +26,12 @@
 		"vi_VN": "GitHub là hệ thống quản lý mã và dự án cũng như là nền tảng mạng xã hội dành cho các nhà phát triển."
 	},
 	"url": "github.com",
-	"version": "2.10.8",
+	"version": "2.10.9",
```
Comment (EncryptedDev) [inline websites/G/GitHub/dist/metadata.json:29]: This should be minor

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
If you are going to add a privacy filter, you should add it to all pages which display private information. For example, you should add this to organization pages, pull requests etc.

**Comment by realtofuine** [comment]
closes #6357

**Comment by realtofuine** [comment]
closes #6355

**Comment by realtofuine** [comment]
resolves #6346

**Review by EncryptedDev — DISMISSED** [review]

File: websites/G/GitHub/presence.ts [hunk 2]
```diff
@@ -89,57 +103,117 @@ presence.on("UpdateData", async () => {
...
 			presenceData.buttons = [{ label: "View Repository", url: href }];
-			if (cover) {
+			if (cover && !privacy) {
 				presenceData.largeImageKey = `https://avatars.githubusercontent.com/u/${
 					document.querySelector<HTMLMetaElement>(
 						'meta[name~="octolytics-dimension-user_id"]'
 					).content
 				}`;
-			}
+			} else presenceData.largeImageKey = "lg";
```
Comment (ImRodry) [inline websites/G/GitHub/presence.ts:112]: this value is already set, this line is useless

File: websites/G/GitHub/presence.ts [hunk 3]
```diff
@@ -69,6 +77,12 @@ presence.on("UpdateData", async () => {
 					.querySelector("span.p-nickname")
 					.textContent.trim();
 			presenceData.buttons = [{ label: "View Profile", url: href }];
+			if (privacy) {
+				presenceData.details = "Viewing a profile";
+				delete presenceData.state;
+				delete presenceData.buttons;
+				break;
+			}
```
Comment (ImRodry) [inline websites/G/GitHub/presence.ts:85]: why set it just to delete it right after?

File: websites/G/GitHub/presence.ts [hunk 4]
```diff
@@ -51,11 +51,19 @@ presence.on("UpdateData", async () => {
...
+		[cover, timestamp, privacy] = await Promise.all([
 			presence.getSetting<boolean>("cover"),
 			presence.getSetting<boolean>("timestamp"),
+			presence.getSetting<boolean>("privacy"),
 		]);
 
+	if (privacy) {
+		delete presenceData.state;
+		presenceData.largeImageKey = "lg";
+		presenceData.startTimestamp = Math.floor(Date.now() / 1000);
```
Comment (ImRodry) [inline websites/G/GitHub/presence.ts:63]: this doesn't work.

File: websites/G/GitHub/presence.ts [hunk 5]
```diff
@@ -181,8 +273,20 @@ presence.on("UpdateData", async () => {
...
+			break;
+		case pathname.includes("/features"):
+			presenceData.details = "Browsing features";
+			if (pathname.includes("copilot"))
+				presenceData.state = "Looking at Github Copilot";
+
 			break;
 		case pathname.includes("/orgs/"):
+			if (privacy) {
+				presenceData.details = "Viewing an organization's people";
```
Comment (ImRodry) [inline websites/G/GitHub/presence.ts:285]: what is this english

**Review by ImRodry — DISMISSED** [review]

**Review by ImRodry — CHANGES_REQUESTED** [review]
Wrong button

**Comment by EncryptedDev** [comment]
Please attach proof individually of each issue linked being resolved.

**Comment by realtofuine** [comment]
> Please attach proof individually of each issue linked being resolved.

ok i did it

its under the screenshots now

**Review by Bas950 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Comment by EncryptedDev** [comment]
Thank you for the PR 🎉

**Pull request merged**